src/components/implementation/torrent/ramfs/ramfs.c

```diff
@@ -134,6 +172,60 @@ tread(spdid_t spdid, td_t td, int cbid, int sz)
 	return ret;
 }
 
+int
+treadp(spdid_t spdid, td_t td, int *off, int *sz)
+{
+	int ret = -1;
+	struct torrent *t;
+	struct fsobj *fso;
+	struct file_data *current, *prev;
+	u32_t total_offset = 0;
+
+	if (tor_isnull(td)) return -EINVAL;
+	
+	LOCK();
+	t = tor_lookup(td);
+	if (!t) ERR_THROW(-EINVAL, done);
+	assert(!tor_is_usrdef(td) || t->data);
+	if (!(t->flags & TOR_READ)) ERR_THROW(-EACCES, done);
+	fso = t->data;
+	
+	//assert(fso->size <= fso->allocated);
+	//assert(t->offset <= fso->size);
+	if (!fso->size) ERR_THROW(0, done);
```

Comment:
Are there any other things we should be checking for here?  Should the assertions be commented out?